### PR TITLE
[Database] Added Database Public Interface UML Class Diagrams

### PR DESCRIPTION
**Pull request by andrewdimmer**

### CODE REVIEW

File: docs/uml-class-diagrams/database01/production/DatabaseGisInterfaceImplementation/DatabaseGisInterfaceImplementation.puml [hunk 1]
```diff
@@ -4,9 +4,11 @@ package edu.oakland.production.database01 {
     class DatabaseGisInterfaceImplementation {
         __
         .. Use Case 1 ..
-
+        {abstract} + void recieveStoreRequest(LocationDataPoint locationDataPoint)
```
Comment (andrewdimmer) [inline docs/uml-class-diagrams/database01/production/DatabaseGisInterfaceImplementation/DatabaseGisInterfaceImplementation.puml]:
```suggestion
        + void recieveStoreRequest(LocationDataPoint locationDataPoint)
```

File: docs/uml-class-diagrams/database01/production/DatabaseGisInterfaceImplementation/DatabaseGisInterfaceImplementation.puml [hunk 2]
```diff
@@ -4,9 +4,11 @@ package edu.oakland.production.database01 {
...
         .. Use Case 2 ..
-
+        {abstract} + void recieveModeRequest(String mode)
+        {abstract} + String recieveNextSatRequest(String currentSat)
```
Comment (andrewdimmer) [inline docs/uml-class-diagrams/database01/production/DatabaseGisInterfaceImplementation/DatabaseGisInterfaceImplementation.puml]:
```suggestion
        + void recieveModeRequest(String mode)
        + String recieveNextSatRequest(String currentSat)
```

File: docs/uml-class-diagrams/database01/production/DatabaseGisManagerImplementation/DatabaseGisManagerImplementation.puml [hunk 3]
```diff
@@ -4,9 +4,10 @@ package edu.oakland.production.database01 {
     class DatabaseGisManagerImplementation {
         __
         .. Use Case 1 ..
-
+        {abstract} + void passStoreRequest(LocationDataPoint locationDataPoint)
```
Comment (andrewdimmer) [inline docs/uml-class-diagrams/database01/production/DatabaseGisManagerImplementation/DatabaseGisManagerImplementation.puml]:
```suggestion
        + void passStoreRequest(LocationDataPoint locationDataPoint)
```

File: docs/uml-class-diagrams/database01/production/DatabaseGisManagerImplementation/DatabaseGisManagerImplementation.puml [hunk 4]
```diff
@@ -4,9 +4,10 @@ package edu.oakland.production.database01 {
...
         .. Use Case 2 ..
-
+        {abstract} + void passModeRequest(String mode)
+        {abstract} + String passNextSatRequest(String currentSat)
```
Comment (andrewdimmer) [inline docs/uml-class-diagrams/database01/production/DatabaseGisManagerImplementation/DatabaseGisManagerImplementation.puml]:
```suggestion
        + void passModeRequest(String mode)
        + String passNextSatRequest(String currentSat)
```

**Review by andrewdimmer — COMMENTED** [review]
In the class implementation, the methods will not be abstract. Can you review and take the changes proposed if you agree with them?

**Review by Endritz — APPROVED** [review]
Makes everything easier to look at and removes abstract from some of the classes so we can create an instance of that class

**Review by camrongrant1 — APPROVED** [review]
Looks good to me!

**Pull request merged**